webhook/resourcesemantics/validation/controller.go

```diff
@@ -40,17 +40,29 @@ func NewAdmissionController(
 	handlers map[schema.GroupVersionKind]resourcesemantics.GenericCRD,
 	wc func(context.Context) context.Context,
 	disallowUnknownFields bool,
+	callbacks ...map[schema.GroupVersionKind]Callback,
 ) *controller.Impl {
 
 	client := kubeclient.Get(ctx)
 	vwhInformer := vwhinformer.Get(ctx)
 	secretInformer := secretinformer.Get(ctx)
 	options := webhook.GetOptions(ctx)
 
+	// This not ideal, we are using a variadic argument to effectively make callbacks optional
+	// This allows this addition to be non-breaking to consumers of /pkg
+	// TODO: once all sub-repos have adoped this, we might move this back to a traditional param.
+	unwrappedCallbacks := map[schema.GroupVersionKind]Callback{}
```

Comment:
I'd keep it simple and panic if len(callbacks) > 1 

I'm suggesting we opt for the panic because fetching the clients above panic as well if the context isn't decorated.

Reply:
Good point. This will be super annoying down the line if anyone ever has > 1